File: Ordenação Memoria/Bolha/C/bolha.c

#include <stdio.h>
void swap(int *i, int *j);
int *bolha(int  arrayDesordenado[] , int tamanho);
int main (){
   int arrayDesordenado[11]={5,10,40,100,1,3,14,20,30,41,29};
   int *arrayOrdenado=bolha(arrayDesordenado,11);
   for(int i =0; i<11;i++){
     printf("%d ",arrayOrdenado[i]);
   }
  return 0;
 }

/* algoritimo de ordenacao*/
int *bolha(int *arrayDesordenado, int tamanho){
  for(int i = (tamanho-1); i>0 ; i--){
    for(int z =0; z<i;z++){
      if(arrayDesordenado[z]>arrayDesordenado[z+1]){
        swap(&arrayDesordenado[z],&arrayDesordenado[z+1]);
      }
    }
  }
  return arrayDesordenado;
}


/* trocar valores no vetor */
void swap(int *i, int *j) {
   int temp = *i;
   *i = *j;
   *j = temp;
}

File: Ordenação Memoria/Bolha/java/Bolha.java
public class Bolha{

  public static void main (String [] agrs){
    int [] arrayDesordenado={10,40,1,74,31,100,30,50,70,20,12,13}; // array desordenado
    int []arrayOrdenado=bolhaShort(arrayDesordenado); // chamamento da funcao
    for(int i =0;i<arrayOrdenado.length;i++){ //mostrar
     System.out.print(arrayOrdenado[i]+"");
    }
  }


/* Algoritimo bolha */
  public static int [] bolhaShort(int []arrayDesordenado){
    for(int i =(arrayDesordenado.length-1);i>0;i--){
      for (int z=0;z<i ;z++ ) {
        if(arrayDesordenado[z]>arrayDesordenado[z+1])
        swap(z,z+1,arrayDesordenado);
      }
    }
    return arrayDesordenado ;
  }


/* metodo swap  */
  public static void swap(int primeiraP , int segundaP, int []arrayParaserTrocado){
    int receber=0;
    receber=arrayParaserTrocado[primeiraP];
    arrayParaserTrocado[primeiraP]=arrayParaserTrocado[segundaP];
    arrayParaserTrocado[segundaP]=receber;
  }

}
